internal/data/LocalSpiceDbContainer.go

```diff
@@ -0,0 +1,198 @@
+		return nil, fmt.Errorf("could not connect to docker: %w", err)
+	}
+
+	pool.MaxWait = 3 * time.Minute
+
+	var (
+		_, b, _, _ = runtime.Caller(0)
+		basepath   = filepath.Dir(b)
+	)
+
+	cmd := []string{"serve-testing", "--skip-release-check=true"}
+
+	var mounts []string
+	if SpicedbSchemaBootstrapFile != "" {
+		cmd = append(cmd, "--load-configs")
+		cmd = append(cmd, "/mnt/spicedb_bootstrap.yaml")
+		mounts = append(mounts, path.Join(basepath, SpicedbSchemaBootstrapFile)+":/mnt/spicedb_bootstrap.yaml")
+	}
+	if SpicedbRelationsBookstrapFile != "" {
+		if SpicedbSchemaBootstrapFile != "" {
+			cmd[len(cmd)-1] = "/mnt/spicedb_bootstrap.yaml,/mnt/spicedb_bootstrap_relations.yaml"
+		} else {
+			cmd = append(cmd, "--load-configs")
+			cmd = append(cmd, "/mnt/spicedb_bootstrap_relations.yaml")
+		}
+		mounts = append(mounts, path.Join(basepath, SpicedbRelationsBookstrapFile)+":/mnt/spicedb_bootstrap_relations.yaml")
+	}
+
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository:   SpicedbImage,
+		Tag:          SpicedbVersion, // Replace this with an actual version
+		Cmd:          cmd,
+		Mounts:       mounts,
+		ExposedPorts: []string{"50051/tcp", "50052/tcp"},
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("could not start spicedb resource: %w", err)
+	}
+
+	port := resource.GetPort("50051/tcp")
+
+	// Give the service time to boot.
+	cErr := pool.Retry(func() error {
+		log.NewHelper(logger).Info("Attempting to connect to spicedb...")
+
+		conn, err := grpc.Dial(
+			fmt.Sprintf("localhost:%s", port),
+			grpc.WithTransportCredentials(insecure.NewCredentials()),
+		)
+		if err != nil {
+			return fmt.Errorf("error connecting to spiceDB: %v", err.Error())
+		}
+
+		client := v1.NewSchemaServiceClient(conn)
+
+		//read scheme we add via mount
+		_, err = client.ReadSchema(context.Background(), &v1.ReadSchemaRequest{})
+
+		return err
+	})
+
+	if cErr != nil {
+		return nil, cErr
+	}
+
+	return &LocalSpiceDbContainer{
+		logger:    logger,
+		port:      port,
+		container: resource,
+		pool:      pool,
+	}, nil
+}
+
+// Port returns the Port the container is listening
+func (l *LocalSpiceDbContainer) Port() string {
+	return l.port
+}
+
+// NewToken returns a new token used for the container so a new store is created in serve-testing
+func (l *LocalSpiceDbContainer) NewToken() (string, error) {
+	buf := make([]byte, 20)
+	if _, err := rand.Read(buf); err != nil {
+		return "", err
+	}
+
+	return base64.StdEncoding.EncodeToString(buf), nil
+}
+
+// WaitForQuantizationInterval needed to avoid read-before-write when loading the schema
+func (l *LocalSpiceDbContainer) WaitForQuantizationInterval() {
+	time.Sleep(10 * time.Millisecond)
+}
+
+// CreateClient creates a new client that connects to the dockerized spicedb instance and the right store
+func (l *LocalSpiceDbContainer) CreateSpiceDbRepository() (*SpiceDbRepository, error) {
+	randomKey, err := l.NewToken()
+	if err != nil {
+		return nil, err
+	}
+
+	var c *conf.Data
```

Comment:
Speaking of passing in a different *conf.Data for tests- could we do it here? It looks like we're passing in an empty one + overrides for all the values. It seems like we could fill in the constructed one with the values- right?

Reply:
Yeah, that code looks a bit confused right now. I think there was an iteration of this code where I wanted/needed to pass in something that was not defined in conf.Data, but they are literally 1:1 at this point, so indeed I think the overrides can go!

Reply:
Ok, so I back-pedalled on those not-very-helpful overrides and now the net changes are minimal. :)